pip/basecommand.py

```diff
@@ -38,6 +39,7 @@ class Command(object):
     name = None
     usage = None
     hidden = False
+    warn_outside_virtualenv = False
```

Comment:
`warn_outside_virtualenv` could be renamed to `warn_no_virtualenv` (shorter) and work well with the `--warn-no-virtualenv` option I suggested in my comment https://github.com/pypa/pip/pull/2401#issuecomment-73096951.